pkg/kv/kvclient/kvcoord/txn_interceptor_pipeliner.go

```diff
@@ -805,10 +813,18 @@ func (tp *txnPipeliner) initializeLeaf(tis *roachpb.LeafTxnInputState) {
 }
 
 // populateLeafFinalState is part of the txnInterceptor interface.
-func (tp *txnPipeliner) populateLeafFinalState(*roachpb.LeafTxnFinalState) {}
+func (tp *txnPipeliner) populateLeafFinalState(tfs *roachpb.LeafTxnFinalState) {
+	// Copy mutable state so access is safe for the caller.
+	lockSpans := tp.lockFootprint.asSlice()
+	tfs.LockSpans = make([]roachpb.Span, len(lockSpans))
+	copy(tfs.LockSpans, lockSpans)
```

Comment:
I don't think the LeafTxn gets reused/extended afterwards, so maybe you could avoid the copy?

Reply:
(If you know a copy is needed, maybe explain why in a comment)